examples/ble-gatt-server/src/main/java/no/nordicsemi/android/ble/ble_gatt_server/GattService.kt

```diff
@@ -0,0 +1,241 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this
+ * software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package no.nordicsemi.android.ble.ble_gatt_server
+
+import android.app.Service
+import android.bluetooth.*
+import android.bluetooth.le.AdvertiseCallback
+import android.bluetooth.le.AdvertiseData
+import android.bluetooth.le.AdvertiseSettings
+import android.content.Context
+import android.content.Intent
+import android.os.Binder
+import android.os.IBinder
+import android.os.ParcelUuid
+import android.util.Log
+import no.nordicsemi.android.ble.BleManager
+import no.nordicsemi.android.ble.BleServerManager
+import no.nordicsemi.android.ble.observer.ServerObserver
+import java.util.*
+
+
+/**
+ * Advertises a Bluetooth LE GATT service and takes care of its requests.
+ */
+class GattService : Service() {
+
+    private var serverManager: ServerManager? = null
+
+    private var bleAdvertiseCallback: BleAdvertiser.Callback? = null
+
+    override fun onCreate() {
+        super.onCreate()
+        enableBleServices()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        disableBleServices()
+    }
+
+    private fun enableBleServices() {
+        serverManager = ServerManager(this)
+        serverManager!!.open()
+
+        bleAdvertiseCallback = BleAdvertiser.Callback()
+
+        val bluetoothManager = getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
+        bluetoothManager.adapter.bluetoothLeAdvertiser?.startAdvertising(
+                BleAdvertiser.settings(),
+                BleAdvertiser.advertiseData(),
+                bleAdvertiseCallback!!
```

Comment:
My understanding is that a peripheral device must continually advertise to be connect-able.

Reply:
Yes. But GATT server can also be used if it's the phone that connects to a peripheral, and the peripheral does service discovery on its own, and both devices use the services of the other side. So advertising is not required for GATT server to be used.

Reply:
Ah, you mean:

1. central connects to peripheral
2. peripheral discovers central's services

But otherwise, a peripheral device must always advertise so that it can be connected with it. I must say, this isn't obvious to the newcomer. :-)